src/Domain/MigrationStep/s150_ProductVariationMigration/ProductVariationMigrator.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\PimMigration\Domain\MigrationStep\s150_ProductVariationMigration;
+
+use Akeneo\PimMigration\Domain\Command\ChainedConsole;
+use Akeneo\PimMigration\Domain\Command\SymfonyCommand;
+use Akeneo\PimMigration\Domain\DataMigration\DataMigrator;
+use Akeneo\PimMigration\Domain\Pim\DestinationPim;
+use Akeneo\PimMigration\Domain\Pim\SourcePim;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Migrates products variations (via IVB and variant-group).
+ *
+ * @author    Laurent Petard <laurent.petard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ */
+class ProductVariationMigrator implements DataMigrator
+{
+    const MAX_VARIANT_AXES = 5;
+
+    const ALLOWED_AXE_TYPES = [
+        'pim_catalog_simpleselect',
+        'pim_reference_data_simpleselect',
+        'pim_catalog_metric',
+        'pim_catalog_boolean',
+    ];
+
+    /** @var InnerVariationTypeMigrator */
+    private $innerVariantTypeMigrator;
+
+    /** @var VariantGroupMigrator */
+    private $variantGroupMigrator;
+
+    /** @var ChainedConsole */
+    private $console;
+
+    /** @var VariantGroupRetriever */
+    private $variantGroupRetriever;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        ChainedConsole $console,
+        InnerVariationTypeMigrator $innerVariantTypeMigrator,
+        VariantGroupMigrator $variantGroupMigrator,
+        VariantGroupRetriever $variantGroupRetriever,
+        LoggerInterface $logger
+    ) {
+        $this->console = $console;
+        $this->innerVariantTypeMigrator = $innerVariantTypeMigrator;
+        $this->variantGroupMigrator = $variantGroupMigrator;
+        $this->variantGroupRetriever = $variantGroupRetriever;
+        $this->logger = $logger;
+    }
+
+    public function migrate(SourcePim $sourcePim, DestinationPim $destinationPim): void
+    {
+        $warnings = [];
+
+        try {
+            $isIvbMigrationDone = $this->migrateInnerVariationBundle($sourcePim, $destinationPim);
+        } catch (InvalidInnerVariationTypeException $exception) {
+            $warnings[] = $exception->getMessage();
```

Comment:
Why not Throw Exception here?

Reply:
Because even if one InnerVariationType is invalid, we have to continue to migrate the others. But we have to keep in memory that there is an invalid one that has not been migrated.

Reply:
No. There can only be a maximum of 2 warnings : one for the variant-group migration, and one for the IVB migration.